tests/cypress/integration/reset-page-ordering.test.js

```diff
@@ -0,0 +1,43 @@
+describe('Test Reset Page Order Change', () => {
+	it('Can reset pages order', () => {
+		cy.login();
+		cy.visit('/wp-admin/edit.php?post_type=page');
+
+		const first = '.wp-list-table tbody tr:nth-child(1)';
+		const second = '.wp-list-table tbody tr:nth-child(2)';
+		const firstText = cy.get(`${first} .row-title`);
+		const secondText = cy.get(`${second} .row-title`);
+
+		// first reorder the pages.
+		cy.get(first).drag(second);
+		// wait for order update done.
+		cy.get(`${second}  .check-column input`).should('exist');
+
+		cy.get(`${first} .row-title`).then($el => {
+			secondText.should('have.text', $el.text());
+		});
+		cy.get(`${second} .row-title`).then($el => {
+			firstText.should('have.text', $el.text());
+		});
+
+		// now reset the page order and verify original values are back.
+		cy.get(`#contextual-help-link`).click();
+		cy.get(`#tab-link-simple_page_ordering_help_tab`).click();
+		cy.get(`#simple-page-ordering-reset`).click();
+		cy.on(`window:confirm`, () => true);
+
+		// wait for the page to reload before checking for reset post order.
+		cy.wait(1000);
+
+		// Re-query row elements that were detached from the DOM during the page reload.
+		const newFirstText = cy.get(`${first} .row-title`);
+		const newSecondText = cy.get(`${second} .row-title`);
+
+		cy.get(`${first} .row-title`).then($el => {
+			newFirstText.should('have.text', $el.text());
```

Comment:
I think this will always be true, as `newFirstText` and `cy.get('${first} .row-title')` refer to the same element.

Reply:
We need to save the original post names so that we can verify their positions after swapping and reverting.
This is something I was trying today, I used `cy.as()` to store the text values and refer them later on for verification:

```JS
describe( 'Test Reset Page Order Change', () => {
	it( 'Can reset pages order', () => {
		cy.login();
		cy.visit('/wp-admin/edit.php?post_type=page');

		const firstRow = '.wp-list-table tbody tr:nth-child(1)';
		const secondRow = '.wp-list-table tbody tr:nth-child(2)';

		// Alias titles as `firstRowText` and `secondRowText` for convenience.
		cy.get( firstRow ).find( '.row-title' ).invoke( 'text' ).as( 'firstRowText' );
		cy.get( secondRow ).find( '.row-title' ).invoke( 'text' ).as( 'secondRowText' );

		// Swap position of `Page 1` with `Page 2`.
		cy.get( firstRow ).drag( secondRow );

		// Verifies if 1st row has title `Page 2`.
		cy.get( firstRow ).find( '.row-title' ).invoke( 'text' ).then( function( text ) {
			expect( text ).to.eq( this.secondRowText );
		} );

		// Verifies if 2nd row has title `Page 1`.
		cy.get( secondRow ).find( '.row-title' ).invoke( 'text' ).then( function( text ) {
			expect( text ).to.eq( this.firstRowText );
		} );

		// Now reset the page order and verify original values are back.
		cy.get( '#contextual-help-link' ).click();
		cy.get( '#tab-link-simple_page_ordering_help_tab' ).click();
		cy.get( '#simple-page-ordering-reset' ).click();
		cy.on( 'window:confirm', () => true );

		// Perform a reload as Cypress won't after window:confirm.
		cy.reload();

		// Verifies if 1st row has title `Page 1`.
		cy.get( firstRow ).find( '.row-title' ).invoke( 'text' ).then( function( text ) {
			expect( text ).to.eq( this.firstRowText );
		} );

		// Verifies if 2nd row has title `Page 2`.
		cy.get( secondRow ).find( '.row-title' ).invoke( 'text' ).then( function( text ) {
			expect( text ).to.eq( this.secondRowText );
		} );
	} );
} );

```